In Python 3.12  xml.etree.ElementTree will raise DeprecationWarning: Testing an element's truth value will raise an exception in future versions

While testing Python 3.12 in CI

https://github.com/scikit-hep/pyhf/blob/adddb0797c564a0158a8e2e69a58ee1f98604bf7/tests/test_export.py#L438-L450

raised 

```pytb
>       assert channel
E       DeprecationWarning: Testing an element's truth value will raise an exception in future versions.  Use specific 'len(elem)' or 'elem is not None' test instead.
```

This comes from https://github.com/python/cpython/issues/83122 which landed in Python 3.12. This should get fixed before Python 3.12 support is added.

From the Python 3.12 docs: https://docs.python.org/3.12/library/xml.etree.elementtree.html#element-objects

> Caution: Elements with no subelements will test as `False`. Testing the truth value of an Element is deprecated and will raise an exception in Python 3.14. Use specific `len(elem)` or `elem is None` test instead.:
>
>
> ```python
> element = root.find('foo')
> 
> if not element:  # careful!
>     print("element not found, or element has no subelements")
> 
> if element is None:
>     print("element not found")
> ```
>
> _Changed in version 3.12_: Testing the truth value of an Element emits [DeprecationWarning](https://docs.python.org/3/library/exceptions.html#DeprecationWarning).

--- a/src/pyhf/writexml.py
+++ b/src/pyhf/writexml.py
@@ -56,7 +56,7 @@ def _export_root_histogram(hist_name, data):
 # https://stackoverflow.com/a/4590052
 def indent(elem, level=0):
     i = "\n" + level * "  "
-    if elem:
+    if elem is not None:
         if not elem.text or not elem.text.strip():
             elem.text = i + "  "
         if not elem.tail or not elem.tail.strip():

--- a/tests/test_export.py
+++ b/tests/test_export.py
@@ -447,7 +447,7 @@ def test_integer_data(datadir, mocker):
     mocker.patch("pyhf.writexml._ROOT_DATA_FILE")
 
     channel = pyhf.writexml.build_channel(spec, channel_spec, {})
-    assert channel
+    assert channel is not None
 
 
 @pytest.mark.parametrize(

